bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngine.java

```diff
@@ -328,18 +328,23 @@ protected synchronized void setRule(String rUID) {
             errMsgs = errMsgs + "\n" + errMessage;
         }
 
-        try {
-            ConnectionValidator.validateConnections(r);
-        } catch (Exception e) {
-            errMsgs = errMsgs + "\n Validation of rule" + r.getUID() + "has failed! " + e.getMessage();
-        }
-
         if (errMsgs == null) {
-            resolveDefaultValues(r);
-            register(r);
 
-            // change state to IDLE
-            setRuleStatusInfo(r.getUID(), new RuleStatusInfo(RuleStatus.IDLE));
+            try {
+                ConnectionValidator.validateConnections(r);
+                resolveDefaultValues(r);
+                register(r);
+                // change state to IDLE
+                setRuleStatusInfo(r.getUID(), new RuleStatusInfo(RuleStatus.IDLE));
+
+            } catch (Exception e) {
```

Comment:
1) I don't like catching exception here, I prefer to change the mehtod ConnectionValidator.validateConnections(r); to return String (whcih is an error message or null in case when validation succeed) . It it stays exception, yes we change to catch IllegalArgumentException
2) Yes it is CONFIGURATION_ERROR